aten/src/ATen/native/cuda/ForeachReduceOp.cu

```diff
@@ -209,10 +209,18 @@ std::vector<Tensor> foreach_tensor_norm_cuda(
         p);
   }
 
+  // correctly assign values to only non-empty slots, as the empty slots should
+  // get skipped
   std::vector<Tensor> result;
   result.reserve(ntensors);
-  for (const auto& i : c10::irange(ntensors)) {
-    result.emplace_back(ret_per_tensor[i]);
+  int i = 0;
+  for (const auto& t : tensors) {
+    if (t.numel() != 0) {
+      result.emplace_back(ret_per_tensor[i]);
+      i++;
+    } else {
+      result.emplace_back(at::zeros({}, options));
```

Comment:
you still need output of correct shape, no?

Reply:
this is the right shape! the only reduce op we have is _foreach_norm and so the output is a 0dim scalar for every tensor.